contracts/WildcardSteward_v2.sol

```diff
@@ -419,77 +470,188 @@ contract WildcardSteward_v2 is Initializable {
+                    uint256 previousCollectionTime
+                 = timeLastCollectedPatron[tokenPatron];
+                uint256 newTimeLastCollected = previousCollectionTime.add(
                     (
-                        (now.sub(timeLastCollectedPatron[currentOwner]))
-                            .mul(deposit[currentOwner])
+                        (now.sub(previousCollectionTime))
+                            .mul(deposit[tokenPatron])
                             .div(patronageOwedByTokenPatron)
                     )
                 );
-
-                timeLastCollected[tokenId] = newTimeLastCollected;
-                _collectLoyaltyPatron(
-                    currentOwner,
-                    newTimeLastCollected.sub(
-                        timeLastCollectedPatron[currentOwner]
-                    )
+                timeLastCollectedPatron[tokenPatron] = newTimeLastCollected;
+                deposit[tokenPatron] = 0;
+                timeSinceLastMint = (
+                    newTimeLastCollected.sub(previousCollectionTime)
                 );
 
+                // The bellow 2 lines are the main difference between this function and the `_collectPatronagePatron` function.
                 tokenAuctionBeginTimestamp[tokenId] = newTimeLastCollected;
-                timeLastCollectedPatron[currentOwner] = newTimeLastCollected;
-                collection = price[tokenId]
-                    .mul(newTimeLastCollected.sub(previousTokenCollection))
-                    .mul(patronageNumerator[tokenId])
-                    .div(1000000000000)
-                    .div(365 days);
-                deposit[currentOwner] = 0;
-
                 _foreclose(tokenId);
-            } else {
-                collection = price[tokenId]
-                    .mul(now.sub(previousTokenCollection))
-                    .mul(patronageNumerator[tokenId])
-                    .div(1000000000000)
-                    .div(365 days);
-
-                _collectLoyaltyPatron(
-                    currentOwner,
-                    now.sub(timeLastCollectedPatron[currentOwner])
-                );
 
-                timeLastCollected[tokenId] = now;
-                timeLastCollectedPatron[currentOwner] = now;
-                currentCollected[tokenId] = currentCollected[tokenId].add(
-                    collection
+                address benefactor = benefactors[tokenId];
+
+                // If the organisation collected their patronage after this token was foreclosed, then record the credit they have been given.
+                if (
+                    benefactorLastTimeCollected[benefactor] >
+                    newTimeLastCollected
+                ) {
+                    benefactorCredit[benefactor] = price[tokenId].mul(
+                        benefactorLastTimeCollected[benefactor].sub(
+                            newTimeLastCollected
+                        )
+                    );
```

Comment:
I see... credit arises when orgs collect patorange but some of the tokens are actually foreclosed and not contributing towards them.

Very important we set a max cap price on assets to avoid attacks against this